test/functionapp.test.ts

```diff
@@ -0,0 +1,125 @@
+    this.timeout(1200 * 1000);
+    const resourceGroupsToDelete: string[] = [];
+    let oldAdvancedCreationSetting: boolean | undefined;
+    let oldProjectLanguage: ProjectLanguage | undefined;
+    const resourceName1: string = getRandomHexString().toLowerCase();
+    const testAccount: TestAzureAccount = new TestAzureAccount();
+
+    suiteSetup(async function (this: IHookCallbackContext): Promise<void> {
+        if (!longRunningTestsEnabled) {
+            this.skip();
+        }
+        this.timeout(120 * 1000);
+        oldProjectLanguage = getGlobalFuncExtensionSetting(projectLanguageSetting);
+        oldAdvancedCreationSetting = <boolean>vscode.workspace.getConfiguration(extensionPrefix).get('advancedCreation');
+        await testAccount.signIn();
+        ext.tree = new AzureTreeDataProvider(FunctionAppProvider, 'azureFunctions.startTesting', undefined, testAccount);
+    });
+
+    suiteTeardown(async function (this: IHookCallbackContext): Promise<void> {
+        if (!longRunningTestsEnabled) {
+            this.skip();
+        }
+        this.timeout(1200 * 1000);
+        await updateGlobalSetting(projectLanguageSetting, oldProjectLanguage);
+        await vscode.workspace.getConfiguration(extensionPrefix).update('advancedCreation', oldAdvancedCreationSetting, vscode.ConfigurationTarget.Global);
+        const client: ResourceManagementClient = getResourceManagementClient(testAccount);
+        for (const resourceGroup of resourceGroupsToDelete) {
+            if (await client.resourceGroups.checkExistence(resourceGroup)) {
+                console.log(`Deleting resource group "${resourceGroup}"...`);
+                await client.resourceGroups.deleteMethod(resourceGroup);
+                console.log(`Resource group "${resourceGroup}" deleted.`);
+            } else {
+                // If the test failed, the resource group might not actually exist
+                console.log(`Ignoring resource group "${resourceGroup}" because it does not exist.`);
+            }
+        }
+        ext.tree.dispose();
+    });
+
+    test('createFunctionApp (Basic)', async () => {
+        const resourceName2: string = getRandomHexString().toLowerCase();
+        resourceGroupsToDelete.push(resourceName2);
+        await vscode.workspace.getConfiguration(extensionPrefix).update('advancedCreation', false, vscode.ConfigurationTarget.Global);
+        await updateGlobalSetting(projectLanguageSetting, ProjectLanguage.JavaScript);
+        ext.ui = new TestUserInput([resourceName2]);
+        await vscode.commands.executeCommand('azureFunctions.createFunctionApp');
+        const client: WebSiteManagementClient = getWebsiteManagementClient(testAccount);
+        const createdApp: WebSiteManagementModels.Site = await client.webApps.get(resourceName2, resourceName2);
+        assert.ok(createdApp);
+    });
+
+    test('createFunctionApp (Advanced)', async () => {
+        resourceGroupsToDelete.push(resourceName1);
```

Comment:
To more throughly test advancedCreation, you should use different names for the function app name, resource group name, and storage account name. Otherwise it's a little too similar to basic create